src/components/node-list/node-list-tree.js

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { makeStyles, withStyles } from '@material-ui/core/styles';
+import TreeView from '@material-ui/lab/TreeView';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+
+import {
+  toggleModularPipelineActive,
+  toggleModularPipelineFilter,
+} from '../../actions/modular-pipelines';
+import { toggleTypeDisabled } from '../../actions/node-type';
+import { getNodeTypes, getNodeTypeIDs } from '../../selectors/node-types';
+import {
+  getModularPipelineIDs,
+  getModularPipelineData,
+} from '../../selectors/modular-pipelines';
+import { getNestedModularPipelines } from './node-list-items';
+import {
+  getGroupedNodes,
+  getNodeSelected,
+  getNodeModularPipelines,
+} from '../../selectors/nodes';
+import { loadNodeData } from '../../actions/nodes';
+import NodeListTreeItem from './node-list-tree-item';
+
+const useStyles = makeStyles({
+  root: {
+    height: 110,
+    flexGrow: 1,
+    maxWidth: 400,
+  },
+});
+
+const StyledTreeView = withStyles({
+  root: {
+    padding: '0 0 0 20px',
+  },
+})(TreeView);
+
+const isModularPipelineType = (type) => type === 'modularPipeline';
+
+const TreeListProvider = ({
+  nodes,
+  nodeSelected,
+  onToggleNodeSelected,
+  searchValue,
+  modularPipelines,
+  modularPipelineIds,
+  nodeModularPipelines,
+  onItemChange,
+  onItemMouseEnter,
+  onItemMouseLeave,
+  nodeTypeIDs,
+}) => {
+  const classes = useStyles();
+
+  const treeData = getNestedModularPipelines({
+    nodes,
+    modularPipelines,
+    nodeSelected,
+    searchValue,
+    modularPipelineIds,
+    nodeModularPipelines,
+    nodeTypeIDs,
+  });
+
+  const onItemClick = (item) => {
+    if (!isModularPipelineType(item.type)) {
+      onToggleNodeSelected(item.id);
+    }
+  };
+
+  const renderTree = (
+    rowData,
+    onItemMouseEnter,
+    onItemMouseLeave,
+    onItemChange,
+    onItemClick
+  ) => {
+    return (
+      <NodeListTreeItem
+        data={rowData}
+        onItemMouseEnter={onItemMouseEnter}
+        onItemMouseLeave={onItemMouseLeave}
+        onItemChange={onItemChange}
+        onItemClick={onItemClick}
+        key={rowData.id}>
+        {rowData.children.length > 0 &&
```

Comment:
Might not need the `rowData.children.length` check here and below?

Reply:
I've deleted this in the new PR